lib/registry.js

```diff
@@ -244,8 +248,12 @@ Registry.prototype.configureModel = function(ModelCtor, config) {
       modelName);
   }
 
+  var baseName = ModelCtor.base.modelName || 'PersistedModel';
+  BaseModel = this.getModel(baseName);
+  var model = BaseModel.extend(modelName, {}, settings);
```

Comment:
You are effectively re-creating the child model here, that looks very wrong to me. I am surprised the tests are passing, considering that they store the original model constructor in  `var MyModel`. I guess this is a quirk of loopback-datasource-juggler, where it merges model configuration when a model with the same name is defined.